SmartDeviceLink/SDLRegisterAppInterface.m

```diff
@@ -42,23 +47,40 @@ - (instancetype)initWithLifecycleConfiguration:(SDLLifecycleConfiguration *)life
                 nightColorScheme:lifecycleConfiguration.nightColorScheme];
 }
 
+- (instancetype)initWithAppName:(NSString *)appName appId:(NSString *)appId fullAppId:(nullable NSString *)fullAppId languageDesired:(SDLLanguage)languageDesired {
+    self = [self initWithAppName:appName languageDesired:languageDesired];
+    if (!self) {
+        return nil;
+    }
+
+    self.fullAppID = fullAppId.length > 0 ? fullAppId : nil;
```

Comment:
This is confusing to me. Why check `length = 0` and set `nil`. If it's `nil`, then the `length == 0` in Obj-C.